torch/csrc/jit/runtime/static/memory_planner.h

```diff
@@ -39,6 +39,38 @@ class StorageGroup {
   std::vector<at::Tensor*> group_{};
 };
 
+// A contiguous buffer of `StorageImpl`s
+class ManagedStorages {
+ public:
+  ManagedStorages();
+
+  ~ManagedStorages();
+
+  void allocate(size_t capacity);
+
+  void deallocate();
+
+  void append(at::StorageImpl* storageImpl);
+
+  at::StorageImpl* operator[](size_t idx);
+
+  const at::StorageImpl* operator[](size_t idx) const;
```

Comment:
wait, why do these return pointers instead of references?

Reply:
No good reason, I've changed them to references